src/test/java/io/jenkins/plugins/analysis/core/columns/IssueCounterITest.java

```diff
@@ -0,0 +1,94 @@
+package io.jenkins.plugins.analysis.core.columns;
+
+import static io.jenkins.plugins.analysis.core.testutil.Assertions.assertThat;
+
+import io.jenkins.plugins.analysis.warnings.Java;
+import java.io.IOException;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.html.DomNode;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import jenkins.model.Jenkins;
+
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerTest;
+
+/**
+ * Integration Test for the Issue counter column.
+ */
+public class IssueCounterITest extends IntegrationTestWithJenkinsPerTest {
+
+    private static final String ISSUE_ELEMENT_XPATH = "/html/body/div[4]/div[2]/div[2]/div[2]/table/tbody/tr[2]/td[8]/a";
+    private static final String JAVAC_ONE_WARNING = "javac_1_warning.txt";
+    private static final String JAVAC_TEN_WARNINGS = "javac_10_warnings.txt";
+    private static final String RESULT_FILE_NAME = "javac.log";
+    private static final String JAVA_FILE_PATTERN = "**/*.log";
+
+    /**
+     * Tests that the issue counter is displayed correctly with 1 issue.
+     * @throws IOException Shouldn't throw.
+     */
+    @Test
+    public void shouldShowIssueCounterEqualToOne() throws IOException {
+        createAndBuildProjectWithFile(JAVAC_ONE_WARNING);
+
+        List<DomNode> element = getRootPage().getByXPath(ISSUE_ELEMENT_XPATH);
+
+        assertThat(Integer.parseInt(element.get(0).asText())).isEqualTo(1);
+    }
+
+    /**
+     * Tests that the issue counter is displayed correctly with 10 issues.
+     * @throws IOException Shouldn't throw.
+     */
+    @Test
+    public void shouldShowIssueCounterEqualToTen() throws IOException {
+        createAndBuildProjectWithFile(JAVAC_TEN_WARNINGS);
+
+        List<DomNode> element = getRootPage().getByXPath(ISSUE_ELEMENT_XPATH);
+
+        assertThat(Integer.parseInt(element.get(0).asText())).isEqualTo(10);
+    }
+
+    /**
+     * Tests that the issue counter is displayed correctly with no issues.
+     * @throws IOException Shouldn't throw.
+     */
+    @Test
+    public void shouldShowIssueCounterEqualToZero() throws IOException {
+        FreeStyleProject project = createFreeStyleProject();
+
+        enableWarnings(project, new Java());
+        scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+
+        List<DomNode> element = getRootPage().getByXPath(ISSUE_ELEMENT_XPATH);
+
+        assertThat(Integer.parseInt(element.get(0).asText())).isEqualTo(0);
+    }
+
```

Comment:
I think the following tests are missing: 
- `-` as result.
- Multiple tools so that the column actually renders the sum. 
- Tool selection via `getJenkins().getInstance().getPrimaryView().getColumns()...`

Reply:
I added a test for the dash case and multiple tools.
I'm not sure what you mean with the tool selection, what exactly should I test with it? If the selection is possible via UI or test it again using the "non-ui" way?

Reply:
I think you can have three parts in your test `shouldShowIssueCounterEqualToSum`. 
1. The existing test
2. Change the visible tools to `java` using the API call (non UI) `getJenkins().getInstance().getPrimaryView().getColumns()...` and validate the number of warnings
3. Change the visible tools to `python` using the API call (non UI) `getJenkins().getInstance().getPrimaryView().getColumns()...` and validate the number of warnings

Maybe it would make sense to change the number of warnings for python or java to 2 so that the assertion correctly works

Reply:
As discussed, this functionality is not working for the test.